### ISSUE TITLE
Training script lr schedulers configuration not taking into account gradient accumulation correctly

### ISSUE THREAD

**Issue by williamberman** [huggingface/diffusers#2632]
### Describe the bug

Re internal discussion: it sounds like accelerate handles prepared lr schedulers step function only being called when gradient updates actually occur. This means that we shouldn't have to change our lr scheduler configuration to artificially stretch out the number of steps based off of the number of gradient accumulation steps.

I'll confirm that my understanding is correct before updating. Using issue as TODO reminder

### Reproduction

n/a

### System Info

n/a

**Comment by williamberman** [comment]
unstale
